lib/maplibre-compose/src/iosMain/kotlin/dev/sargunv/maplibrecompose/compose/offline/OfflineRegionManager.kt

```diff
@@ -0,0 +1,115 @@
+package dev.sargunv.maplibrecompose.compose.offline
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import cocoapods.MapLibre.MLNOfflinePack
+import cocoapods.MapLibre.MLNOfflineStorage
+import dev.sargunv.maplibrecompose.core.util.KVObserverProtocol
+import dev.sargunv.maplibrecompose.core.util.toNSData
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+import platform.Foundation.NSKeyValueChangeSetting
+import platform.Foundation.NSKeyValueObservingOptionNew
+import platform.Foundation.addObserver
+import platform.darwin.NSObject
+
+@Composable
+public actual fun rememberOfflineRegionManager(): OfflineRegionManager {
+  return IosOfflineRegionManager
+}
+
+internal object IosOfflineRegionManager : OfflineRegionManager, NSObject(), KVObserverProtocol {
+
+  private val impl = MLNOfflineStorage.sharedOfflineStorage
+
+  private val regionsState = mutableStateOf(emptySet<OfflineRegion>())
+
+  override val regions
+    get() = regionsState.value
+
+  // TODO observe progress notifications and update region states:
+  // https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre/mlnofflinepackprogresschangednotification
+  // https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre/mlnofflinepackmaximummapboxtilesreachednotification
+  // https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre/mlnofflinepackerrornotification
+
+  init {
+    impl.addObserver(this, "packs", NSKeyValueObservingOptionNew, null)
+  }
+
+  override fun observeValueForKeyPath(
+    keyPath: String?,
+    ofObject: Any?,
+    change: Map<Any?, *>?,
+    context: kotlinx.cinterop.CPointer<out kotlinx.cinterop.CPointed>?,
+  ) {
+    if (
+      ofObject == impl &&
+        keyPath == "packs" &&
+        change != null &&
+        change.contains(NSKeyValueChangeSetting)
```

Comment:
Note to self: I suspect this is wrong; gotta read up on https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/KeyValueObserving/Articles/KVOBasics.html